reactjs_yh/src/U1/hw4/originalNumber.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from "react";
+
+export default class filter extends Component {
+  state = { input: 0, output: "none" };
+  handleChange = (e) => {
+    e.preventDefault();
+    let input = +e.target.value;
+    if (!isNaN(input)) {
+      this.setState({ input: e.target.value });
+      if (e.target.value.at(-1) === "1") {
```

Comment:
since you have already set a local variable `input` why not use it through the function? also the logic get something wrong, e.g., if user input 12 it should return `12th` rather than `12nd` as you indicate here.